### ISSUE TITLE
epub support

### ISSUE THREAD

**Issue by yasirmturk** [Marxon13/M13PDFKit#9]
any possibility?

**Comment by Marxon13** [comment]
Possibly. I could do it. But, does it fit in with a PDF reader? If I do, I might make a new project based off of this one.

**Comment by simalexan** [comment]
@Marxon13 agreed. imho, making a separate project with an epub one is a good idea.

**Comment by Marxon13** [comment]
Or... What about a converter that will convert the ePub into PDF? Hm... I have no idea how hard that would be to implement. The UI's of the PDF and ePub viewers will be very similar, but nearly impossible to separate from the underlying format. I want to re-use as much code as possible.

**Comment by simalexan** [comment]
as an additional feature, for rendering epub as PDF?

The problem with the PDF is that its a visual document format
 [epub standard](http://www.hxa.name/articles/content/epub-guide_hxa7241_2007.html) is more a semantic one - its XML based
tbh, it appears to me as well, that it would be much easier to make a separate project that only handles epub, than to make a converter from epub to pdf.

Maybe we could make separate document rendering engines (the current for PDFs and second one for epubs) and integrate into one viewer, with an option that user can specify which types of documents to support as parameters or something like that? But that again, it would require two (or even 3) separate projects

**Comment by Marxon13** [comment]
I can see two ways of doing this.

- One is creating a separate project for viewing ePub documents.
    - The issues I have with this is that any UI changes or fixes will probably apply across the PDF and ePub projects. And it will be a bit of a hassle to keep in sync.
- The other is to create a backend for ePub, and hook it up to the UI.
    - My concern with this is that I want to add a lot of PDF-centric features, that ePub probably doesn't support. And that there are things in ePub that PDF does't support.

I just took a look at the ePub feature list, and a separate project is probably in order. I might abstract out a bit of the UI into a third project and import that into both, but having a separate project is probably best.

**Comment by simalexan** [comment]
@Marxon13 count me in for the epub project
